MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -665,251 +587,409 @@ private StreamInfo BuildVideoItem(MediaSourceInfo item, VideoOptions options)
+            var (canDirectPlayStream, transcodeReasons) = TryBuildVideoStreamForDirectPlay(item, options, videoStream, audioStream, subtitleStream, outputStream);
+            if (canDirectPlayStream)
+            {
+                return outputStream;
+            }
 
-            var transcodeReasons = new List<TranscodeReason>();
+            // Try to find a DirectPlayProfile that supports source video stream, so video can just be remuxed.
+            var canNonVideoTranscodeStream = TryBuildVideoStreamForNonVideoTranscode(transcodeReasons, item, options, videoStream, audioStream, subtitleStream, outputStream);
+            if (canNonVideoTranscodeStream)
+            {
+                return outputStream;
+            }
 
-            if (isEligibleForDirectPlay || isEligibleForDirectStream)
+            // Fallback to a full transcode of the stream.
+            var canFullTranscodeStream = TryBuildVideoStreamForFullTranscode(transcodeReasons, item, options, videoStream, audioStream, subtitleStream, outputStream);
+            if (canFullTranscodeStream)
             {
-                // See if it can be direct played
-                var directPlayInfo = GetVideoDirectPlayProfile(options, item, videoStream, audioStream, isEligibleForDirectStream);
-                var directPlay = directPlayInfo.Item1;
+                return outputStream;
+            }
 
-                if (directPlay != null)
-                {
-                    playlistItem.PlayMethod = directPlay.Value;
-                    playlistItem.Container = NormalizeMediaSourceFormatIntoSingleContainer(item.Container, options.Profile, DlnaProfileType.Video);
+            return null;
+        }
 
-                    if (subtitleStream != null)
+        private static int GetDefaultAudioBitrate(string audioCodec, int? audioChannels)
+        {
+            if (!string.IsNullOrEmpty(audioCodec))
+            {
+                // Default to a higher bitrate for stream copy
+                if (string.Equals(audioCodec, "aac", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(audioCodec, "mp3", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(audioCodec, "ac3", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(audioCodec, "eac3", StringComparison.OrdinalIgnoreCase))
+                {
+                    if ((audioChannels ?? 0) < 2)
                     {
-                        var subtitleProfile = GetSubtitleProfile(item, subtitleStream, options.Profile.SubtitleProfiles, directPlay.Value, _transcoderSupport, item.Container, null);
-
-                        playlistItem.SubtitleDeliveryMethod = subtitleProfile.Method;
-                        playlistItem.SubtitleFormat = subtitleProfile.Format;
+                        return 128000;
                     }
 
-                    return playlistItem;
+                    return (audioChannels ?? 0) >= 6 ? 640000 : 384000;
```

Comment:
This should be the same
```suggestion
                    return audioChannels >= 6 ? 640000 : 384000;
```